controller/engine_controller.go

```diff
@@ -426,6 +431,61 @@ func (ec *EngineController) DeleteInstance(obj interface{}) error {
 	return nil
 }
 
+func (ec *EngineController) deleteInstanceWithCLIAPIVersionOne(e *longhorn.Engine) (err error) {
+	isCLIAPIVersionOne, err := ec.ds.IsEngineImageCLIAPIVersionOne(e.Status.CurrentImage)
+	if err != nil {
+		return err
+	}
+
+	if isCLIAPIVersionOne {
+		pod, err := ec.podLister.Pods(ec.namespace).Get(e.Name)
```

Comment:
Don't need the informer. Use API directly for the old version - it shouldn't last long.